python/ray/serve/schema.py

```diff
@@ -545,6 +546,34 @@ def get_empty_schema_dict() -> Dict:
         return {"applications": []}
 
 
+@PublicAPI(stability="alpha")
+class ReplicaDetails(BaseModel, extra=Extra.forbid):
+    replica_id: str = Field(
+        description=(
+            "Unique ID for the replica. By default, this will be "
+            '"<deployment name>#<replica suffix>", where the replica suffix is a '
+            "randomly generated unique string."
+        )
+    )
+    state: ReplicaState = Field(description="Current state of the replica.")
+    pid: Optional[int] = Field(description="PID of the replica actor process.")
+    actor_name: str = Field(description="Name of the replica actor.")
+    actor_id: Optional[str] = Field(description="ID of the replica actor.")
+    node_id: Optional[str] = Field(
+        description="ID of the node that the replica actor is running on."
+    )
+    node_ip: Optional[str] = Field(
+        description="IP address of the node that the replica actor is running on."
+    )
+    start_time_s: float = Field(
```

Comment:
Is there a reason this field isn't always the time when the replica actor started? Could the replica store its start time locally and send it back to the controller after the controller recovers?

Reply:
Yeah I'm not sure why we do this, but the start time is reset upon recovery [here](https://github.com/ray-project/ray/blob/master/python/ray/serve/_private/deployment_state.py#L743).

I'm not sure what you mean by locally - what I could think of was storing the start time on each replica actor, then fetching it either after starting the replica or after the controller recovers.

Reply:
> Yeah I'm not sure why we do this, but the start time is reset upon recovery [here](https://github.com/ray-project/ray/blob/master/python/ray/serve/_private/deployment_state.py#L743).

Interesting– I'm not sure either. @edoakes do you know why we reset start time?

> I'm not sure what you mean by locally - what I was considering was storing the start time on each replica actor, then fetching it either after starting the replica or after the controller recovers. Is this also what you were referring to?

Yep, that's what I mean too.

Reply:
We reset start time because the start time is lost when the controller crashes (it's stored in memory).

We could fix this by instead storing the start_time on the replica and returning it w/ the metadata, but it seems like a P1 follow up to me.